ipatests/test_xmlrpc/test_host_plugin.py

```diff
@@ -337,6 +341,98 @@ def test_add_host_with_null_password(self, host):
+                                   self.generate_hostname(MAXHOSTNAMELEN)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+        testhost.ensure_missing()
+
+    def test_try_hostname_length_above_maxhostnamelimit(self):
+        """Try to create host with hostname length above
+        hostnamelength limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN + 1)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        host_create_command = testhost.make_create_command()
+        with raises_exact(errors.ValidationError(
+                name=u'hostname',
+                error=u'can be at most {} characters'.format(
+                    MAXHOSTNAMELEN))):
+            host_create_command()
+        testhost.ensure_missing()
+
+    def test_try_hostname_length_bellow_maximum(self):
+        """Try to create host with valid hostname. Should pass"""
+        valid_length = MAXHOSTNAMELEN // 2
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(valid_length)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN)
+        testhost.delete()
+        host_create_command = testhost.make_create_command()
+        host_create_command()
+        testhost.ensure_missing()
+
+    def test_raise_limit_above_and_try_hostname_len_above_limit(self):
+        """Raise limit above default and try to create host with hostname
+        length above the new-set limit. Should fail"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN * 2)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+        testhost.ensure_missing()
+        host_create_command = testhost.make_create_command()
+        with raises_exact(errors.ValidationError(
+                name=u'hostname',
+                error=u'invalid domain-name: DNS label cannot '
+                      u'be longer that {} characters'.format(
+                          MAXHOSTNAMELEN - 1))):
+            host_create_command()
+        testhost.ensure_missing()
+
+    def test_raise_limit_and_try_valid_len_hostname(self):
+        """Raise limit above default and test hostname with length
+        in between default 64 and the new value. Should pass"""
+        testhost = HostTracker(name=u'testhost',
+                               fqdn=u'{}'.format(
+                                   self.generate_hostname(MAXHOSTNAMELEN + 1)))
+        self.modify_config_maxhostname(testhost, MAXHOSTNAMELEN * 2)
+        testhost.delete()
+        host_create_command = testhost.make_create_command()
+        host_create_command()
+        testhost.ensure_missing()
```

Comment:
Instead of

        host_create_command = testhost.make_create_command()
        host_create_command()

you can use

        testhost.create()

This is equivalent but will also update the tracker's internal state to exists=True, and make the subsequent call to ensure_missing work.